funcbench/main.go

```diff
@@ -248,38 +246,39 @@ func startBenchmark(
 	bench.logger.Println("Assuming comparing with target (clean workdir will be checked.)")
 
 	// Execute benchmark A.
-	newResult, err := bench.exec(ctx, wt.Filesystem.Root(), ref.Hash())
+	newResult, err := bench.exec(wt.Filesystem.Root(), ref.Hash())
 	if err != nil {
 		return nil, errors.Wrapf(err, "execute benchmark for A: %v", ref.Name().String())
 	}
 
+	// TODO move the following part before 'Execute benchmark B.' into a function Benchmarker.switchToWorkTree.
 	// Best effort cleanup and checkout new worktree.
 	if err := os.RemoveAll(cmpWorkTreeDir); err != nil {
 		return nil, errors.Wrapf(err, "delete worktree at %s", cmpWorkTreeDir)
 	}
 
 	// TODO (geekodour): switch to worktree remove once we decide not to support git<2.17
-	if _, err := bench.c.exec(ctx, "git", "worktree", "prune"); err != nil {
+	if _, err := bench.c.exec("git", "worktree", "prune"); err != nil {
 		return nil, errors.Wrap(err, "worktree prune")
 	}
 
 	bench.logger.Println("Checking out (in new workdir):", cmpWorkTreeDir, "commmit", targetCommit.String())
-	if _, err := bench.c.exec(ctx, "git", "worktree", "add", "-f", cmpWorkTreeDir, targetCommit.String()); err != nil {
+	if _, err := bench.c.exec("git", "worktree", "add", "-f", cmpWorkTreeDir, targetCommit.String()); err != nil {
 		return nil, errors.Wrapf(err, "checkout %s in worktree %s", targetCommit.String(), cmpWorkTreeDir)
 	}
 
 	// Execute benchmark B.
-	oldResult, err := bench.exec(ctx, cmpWorkTreeDir, targetCommit)
+	oldResult, err := bench.exec(cmpWorkTreeDir, targetCommit)
 	if err != nil {
 		return nil, errors.Wrapf(err, "execute benchmark for B: %v", env.CompareTarget())
 	}
 
 	// Compare B vs A.
-	cmps, err := bench.compareBenchmarks(oldResult, newResult)
+	tables, err := bench.compareBenchmarks(oldResult, newResult)
 	if err != nil {
 		return nil, errors.Wrap(err, "comparing benchmarks")
 	}
-	return cmps, nil
+	return tables, nil
```

Comment:
Before returning could check if `tables` is empty and return the error that the benchmark might be missing in one or both of the branches. This will also resolve https://github.com/prometheus/test-infra/issues/398 if i am not wrong.